src/tls/openssl.c

```diff
@@ -133,14 +133,60 @@ static void tls_context_destroy(void *ctx_backend)
     flb_free(ctx);
 }
 
+#ifdef _MSC_VER
+static int windows_load_system_certificates(struct tls_context *ctx)
+{
+    int ret;
+
+    HANDLE h;
+    PCCERT_CONTEXT cert = NULL;
+    X509_STORE *store = SSL_CTX_get_cert_store(ctx->ctx);
+    X509 *x509;
+
+    h = CertOpenSystemStoreA(0, "Root");
+    if (h == NULL) {
+        flb_error("[tls] Cannot open cert store: %i", GetLastError());
+        return -1;
+    }
+
+    while (cert = CertEnumCertificatesInStore(h, cert)) {
+        if (cert->dwCertEncodingType & X509_ASN_ENCODING) {
+            // Decode the certificate into X509 struct
+            // `buf` is necessary per openssl docs because the pointer is incremented
+            const unsigned char *buf = cert->pbCertEncoded;
```

Comment:
Could you move the definition of this variable to the beginning of the function?

Reply:
👍🏻